pkg/lightning/mydump/loader.go

```diff
@@ -532,12 +536,24 @@ func (s *mdLoaderSetup) constructFileInfo(ctx context.Context, path string, size
 		}
 		s.tableDatas = append(s.tableDatas, info)
 	case SourceTypeParquet:
-		parquestDataSize, err2 := SampleParquetDataSize(ctx, info.FileMeta, s.loader.GetStore())
-		if err2 != nil {
-			logger.Error("fail to sample parquet data size", zap.String("category", "loader"),
-				zap.String("schema", res.Schema), zap.String("table", res.Name), zap.Stringer("type", res.Type), zap.Error(err2))
-		} else {
-			info.FileMeta.RealSize = parquestDataSize
+		tableName := info.TableName.String()
+		if s.sampledParquetRowSizes[tableName] == 0 {
+			s.sampledParquetRowSizes[tableName], err = SampleParquetRowSize(ctx, info.FileMeta, s.loader.GetStore())
+			if err != nil {
+				logger.Error("fail to sample parquet row size", zap.String("category", "loader"),
+					zap.String("schema", res.Schema), zap.String("table", res.Name),
+					zap.Stringer("type", res.Type), zap.Error(err))
+			}
+		}
+		if s.sampledParquetRowSizes[tableName] != 0 {
+			totalRowCount, err2 := ReadParquetFileRowCountByFile(ctx, s.loader.GetStore(), info.FileMeta)
```

Comment:
this function runs for each file and in serial. and we will do it again in `importTables`, have you tested how long this takes?

for parquet, RealSize is only used in precheck/estimate-compaction-threshold/progress-related, so maybe revert https://github.com/pingcap/tidb/pull/46984 completely or disable it on default

Reply:
In this pr, the `ReadParquetFileRowCountByFile` only run once for each table.

Reply:
With this PR, `ReadParquetFileRowCountByFile` will no longer affect import time.

Reply:
`SampleParquetRowSize` runs once if sampledParquetRowSizes is 0 for the table, it's not zero later. So for later files of the same table `ReadParquetFileRowCountByFile` is called

Reply:
Sorry, I misunderstood `ReadParquetFileRowCountByFile` at first

Reply:
It looks like we need to cache the row count for parquet

Reply:
PTAL～